### PR TITLE
Update to v0.1.6

### PR DESCRIPTION
**Pull request by janaHoffmann1**
Changed
- Change output format from csv to tsv.
- Increase the landscape value for all non-repetitive peptides, not just at the first occurrence.
- Plot core epitope lengths instead of the length of the entire consensus epitope.
- Change the calculation of the total intensity of the evidence file by counting each peptide intensity only once.
- Adjust column output order to column input order.

Fixed
- Remove modification of peptides for peptide length histogram.
- Adjust the epicore intensity plot range to make sure that all epitopes are included.
- Remove redundant intensity information.
- Fix aggregation method for peptides with the same sequence but different properties.
- Fix mapping of cores and input peptides for peptide sequences that occur multiple times in the evidence file.
- Fix proteome_occurrence column for peptides that are part of a repetitive region.

### CODE REVIEW

File: epicore_utils/modules/map_result.py [hunk 1]
```diff
@@ -36,6 +37,27 @@ def read_entire_id_output(id_output: str) -> pd.DataFrame:
         raise Exception('The file type of your evidence file is not supported. Please use an evidence file that has one of the following file types: csv, tsv, xlsx')
     return peptides_df
 
+def aggregation_strategy(series, delimiter):
```
Comment (jonasscheid) [inline epicore_utils/modules/map_result.py]: Add type hinting

Comment (jonasscheid) [reply epicore_utils/modules/map_result.py]: Rename function to what it does.

File: epicore_utils/modules/map_result.py [hunk 2]
```diff
@@ -106,21 +131,22 @@ def map_pep_core(evidence_file: str, protein_df: pd.DataFrame, seq_column: str,
...
-    # create dictionary for aggregation 
-    all_cols = evidence_file_df.columns
-    agg_dict = {}
-    for col in all_cols:
-        if col in [protacc_column,'whole_epitopes_all','consensus_epitopes_all', 'core_epitopes_intensity_all', 'relative_core_intensity_all', 'proteome_occurrence']:
-            agg_dict[col] = lambda x: ','.join(x)
-        else:
-            agg_dict[col] = 'first'
-    grouped_evidence_file_df = evidence_file_df.groupby([seq_column], as_index=False).agg(agg_dict)                                                                                                            
+    group_cols = [col for col in evidence_file_df.columns if col not in [protacc_column, 'start', 'end', 'whole_epitopes_all','consensus_epitopes_all', 'core_epitopes_intensity_all', 'relative_core_intensity_all', 'proteome_occurrence']]
```
Comment (jonasscheid) [inline epicore_utils/modules/map_result.py:135]: Describe in comment what is happening here

File: tests/epicore_test.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+start_column = ''
+end_column = ''
+delimiter = ';'
+mod_pattern = ''
+min_overlap = 7
+max_step_size = 5
+min_epi_length = 10
+
+# run epicore on test files
+proteome_dict = proteome_to_dict(proteome_path)
```
Comment (jonasscheid) [inline tests/epicore_test.py]: DRY: Dont repeat yourself. write a function that returns pep_cores_mapping with different parameters and call it for test case one and two

**Review by jonasscheid — CHANGES_REQUESTED** [review]
Very good that you included tests and also edge-cases! Makes the code more reliable. Only small things that need to be changed

**Review by jonasscheid — APPROVED** [review]
